waspc/data/Generator/templates/sdk/wasp/server/auth/hooks.ts

```diff
@@ -0,0 +1,29 @@
+import { type ProviderId } from '../../auth/utils.js'
+import { type PrismaClient } from '@prisma/client'
+
+export type AuthHookFn =
```

Comment:
I wonder what is the point of this type.
You say it is one of three things. But you have no way to distinguish between those three (no discriminant). So this makes sense only if you don't care which type it is, it is all the same. But then why do you even have 3 types below, why not collapse it all into a single type that is `function (context: { providerId, prisma }) ...` and that is it?
I still have to read more code, maybe it will make sense, but right now it looks like this split into three hook types that are then unified into one hook type would be the same if it was just one hook type from start.